app/liquid_tags/github_tag/github_readme_tag.rb

```diff
@@ -3,71 +3,79 @@
 class GithubTag
   class GithubReadmeTag
     PARTIAL = "liquids/github_readme".freeze
+    GITHUB_DOMAIN_REGEXP = %r{.*github.com/}.freeze
+    OPTION_NO_README = "no-readme".freeze
+    VALID_OPTIONS = [OPTION_NO_README].freeze
 
-    attr_reader :client, :content, :options, :readme_html
-
-    def initialize(link)
-      parsed_link = parse_link(link)
-      @options = parse_options(link)
-      @client = Octokit::Client.new(access_token: token)
-      @content = client.repository(parsed_link)
-      @readme_html = fetch_readme(parsed_link)
+    def initialize(input)
+      @repository_path, @options = parse_input(input)
     end
 
     def render
+      content = Github::Client.repository(repository_path)
+
+      if show_readme?
+        readme_html = fetch_readme(repository_path)
+      end
+
       ActionController::Base.new.render_to_string(
         partial: PARTIAL,
         locals: {
           content: content,
-          show_readme: show_readme? && readme_html.present?,
+          show_readme: readme_html.present?,
           readme_html: readme_html
         },
       )
+    rescue Github::Errors::NotFound, Github::Errors::InvalidRepository
+      raise_error
     end
 
     private
 
-    def parse_link(link)
-      link = sanitize_link(link)
-      parsed_link = link.split(" ").first.delete(" ")
-      raise_error if parsed_link.split("/").length > 2
+    attr_reader :repository_path, :options, :content
 
-      parsed_link
-    end
+    def parse_input(input)
+      sanitized_input = sanitize_input(input)
 
-    def valid_option(option)
-      option.match(/no-readme/)
-    end
+      path, *options = sanitized_input.split(" ")
 
-    def parse_options(link)
-      opts = sanitize_link(link)
-      _, *options = opts.split(" ")
+      validate_options!(*options)
 
-      validated_options = options.map { |option| valid_option(option) }.reject(&:nil?)
-      raise StandardError, "GitHub tag: `#{link}`: Invalid option - did you mean `no-readme`?" unless options.empty? || validated_options.any?
+      path.gsub!(%r{/\z}, "") # remove optional trailing forward slash
+      repository_path = URI.parse(path)
+      repository_path.query = repository_path.fragment = nil
 
-      options
+      [repository_path.normalize.to_s, options]
+    end
+
+    def validate_options!(*options)
+      return if options.empty?
+      return if options.all? { |o| VALID_OPTIONS.include?(o) }
+
+      message = "GitHub tag: invalid options: #{options - VALID_OPTIONS} - supported options: #{VALID_OPTIONS}"
+      raise StandardError, message
```

Comment:
Same as above, will these end up bubbling up to Honeybadger? If so, we may want to consider only showing them to the user rather than pinging us when they happen since there is likely not much we can do.

Reply:
They are shown to the user, I didn't change the current behavior. The `GithubReadmeTag` is rendered in here:

https://github.com/thepracticaldev/dev.to/blob/06c8be370f5c6e6b341ef0ae51617ade7b804f32/app/liquid_tags/github_tag.rb#L18-L27

Liquid errors are shown to the user already:

**master**

![Screenshot_2020-05-07 New Post - DEV Community 👩‍💻👨‍💻(2)](https://user-images.githubusercontent.com/146201/81305404-93e8ad80-907e-11ea-8bae-dfb63b27a450.png)

![Screenshot_2020-05-07 New Post - DEV Community 👩‍💻👨‍💻(3)](https://user-images.githubusercontent.com/146201/81305468-a662e700-907e-11ea-82fa-a2d44853e1a5.png)

**this branch**

![Screenshot_2020-05-07 New Post - DEV(local) Community 👩‍💻👨‍💻](https://user-images.githubusercontent.com/146201/81305876-21c49880-907f-11ea-93e7-c2106f80a954.png)

![Screenshot_2020-05-07 New Post - DEV(local) Community 👩‍💻👨‍💻(1)](https://user-images.githubusercontent.com/146201/81305950-39038600-907f-11ea-8121-a51fc489a3fa.png)

Reply:
Why `StandardError`, not super cool custom error?
Though, as you haven't changed the logic, this doesn't need to be changed in this pr.

Reply:
@lightalloy I don't know, I'd like to fix that eventually. All liquid tags use `StandardError` and that's bothering me a little :D